From 093d328070296cc47d4da339842834907ab8e718 Mon Sep 17 00:00:00 2001
From: xueminsu <xuemin.su@intel.com>
Date: Tue, 30 Oct 2012 13:34:10 +0800
Subject: [PORT FROM MAIN] AudioTrack: fix tombstone in AudioTrack::processAudioBuffer

BZ: 63736

This fixes the tombstone in AudioTrack::processAudioBuffer.
We should createTrack_l first and if we succeed, we create
the AudioTrackThread to run.
Otherwise tombstone happens with NULL mAudioTrack and NULL mCblk.

Change-Id: Ic481ee2ec10d7aba42e3874ce53a52bb659ee17b
Orig-Change-Id: Idee37a628a1fb93141ef0e9bb7043dd9b8d9f639
Signed-off-by: xueminsu <xuemin.su@intel.com>
Signed-off-by: He Bo <bo.he@intel.com>
Reviewed-on: http://android.intel.com:8080/78773
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libmedia/AudioTrack.cpp |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/media/libmedia/AudioTrack.cpp b/media/libmedia/AudioTrack.cpp
index 1bd839f..ec8c12f 100644
--- a/media/libmedia/AudioTrack.cpp
+++ b/media/libmedia/AudioTrack.cpp
@@ -286,11 +286,6 @@ status_t AudioTrack::set(
     mFlags = flags;
     mCbf = cbf;
 
-    if (cbf != NULL) {
-        mAudioTrackThread = new AudioTrackThread(*this, threadCanCallJava);
-        mAudioTrackThread->run("AudioTrack", ANDROID_PRIORITY_AUDIO, 0 /*stack*/);
-    }
-
     // create the IAudioTrack
     status_t status = createTrack_l(streamType,
                                   sampleRate,
@@ -308,6 +303,11 @@ status_t AudioTrack::set(
         return status;
     }
 
+    if (cbf != NULL) {
+        mAudioTrackThread = new AudioTrackThread(*this, threadCanCallJava);
+        mAudioTrackThread->run("AudioTrack", ANDROID_PRIORITY_AUDIO, 0 /*stack*/);
+    }
+
     mStatus = NO_ERROR;
 
     mStreamType = streamType;
-- 
1.7.4.1

